CMakeLists.txt

```diff
@@ -0,0 +1,43 @@
+cmake_minimum_required(VERSION 3.6)
+
+project(Sc2LadderServer)
+
+# Specify output directories.
+set(EXECUTABLE_OUTPUT_PATH ${PROJECT_BINARY_DIR}/bin)
+
+# Build with c++14 support, required by sc2api.
+set(CMAKE_CXX_STANDARD 14)
+
+# Allow creating filters for projects in visual studio.
+set_property(GLOBAL PROPERTY USE_FOLDERS ON)
+
+# Disable building of examples in the sc2api submodule.
+set(BUILD_API_EXAMPLES OFF CACHE INTERNAL "" FORCE)
+
+# Disable building of tests in the sc2api submodule.
+set(BUILD_API_TESTS OFF CACHE INTERNAL "" FORCE)
+
+add_subdirectory("s2client-api")
+add_subdirectory("src")
+
+# set Sc2LadderServer as startup project
+set_property(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} PROPERTY VS_STARTUP_PROJECT Sc2LadderServer)
+
+# set folders/subfolders
+# there might be a better way to do this than
+# manually setting all these, but it works...
+set_target_properties(sc2api PROPERTIES FOLDER sc2api)
+set_target_properties(sc2lib PROPERTIES FOLDER sc2api)
+set_target_properties(sc2protocol PROPERTIES FOLDER sc2api)
+set_target_properties(sc2utils PROPERTIES FOLDER sc2api)
+
+set_target_properties(sc2renderer PROPERTIES FOLDER sc2api/utilities)
+
+set_target_properties(civetweb-c-library PROPERTIES FOLDER sc2api/contrib)
+set_target_properties(libprotobuf PROPERTIES FOLDER sc2api/contrib)
+set_target_properties(libprotobuf-lite PROPERTIES FOLDER sc2api/contrib)
+set_target_properties(libprotoc PROPERTIES FOLDER sc2api/contrib)
+set_target_properties(protoc PROPERTIES FOLDER sc2api/contrib)
+set_target_properties(ipv6-parse PROPERTIES FOLDER sc2api/contrib)
+set_target_properties(SDL2main PROPERTIES FOLDER sc2api/contrib)
+set_target_properties(SDL2-static PROPERTIES FOLDER sc2api/contrib)
```

Comment:
If I not mistaken all these properties should be set inside s2client-api's CMakeLists.txt, not here.
Also what is the purpose of this?

Reply:
Just to clarify; without it, it ends up like so:
![image](https://user-images.githubusercontent.com/14326310/40708526-d0e758dc-6432-11e8-9c74-5c9848c09ed0.png)

Reply:
Yeh, I think keeping the s2client-api as it current is would definately be preferable.  That way its seperated in a seperate folder, and we don't have to worry about it if we don't want to.  Is there any extra work involved if the api changes?  that would be my only worry, but for the moment, I think this is better

Reply:
On the second thought probably I am wrong. An API developer highly likely wants to see these folders.
Since I don't use VS it is not a problem to me so we can keep it here but better to hide under an if. It is Windows-related anyway.

Reply:
> Is there any extra work involved if the api changes?

If I understand you correctly: If extra projects are added to s2client-api, we could need to configure them additionally in the same way.
I would prefer a way of saying to cmake simply: "take s2client-api's generated folder tree and put it all under an 's2client-api' folder.

> but better to hide under an if. It is Windows-related anyway.

From what I read, it's a general IDE thing, but Visual Studio is the only one I've seen mention of it affecting. I'm happy to put it under an if.

Reply:
I vote to postpone this change. We can discuss and merge it separately.

Reply:
I'm not too fussed either way. You guys are welcome to decide.

Reply:
I'm not going to merge this in today, I'll tak to you about it tomorrow if you're about

Reply:
No probs - chat when you're around.